pkg/distributor/distributor_ring.go

```diff
@@ -22,10 +22,10 @@ type RingConfig struct {
 	HeartbeatTimeout time.Duration `yaml:"heartbeat_timeout,omitempty"`
 
 	// Instance details
-	InstanceID             string   `yaml:"instance_id"`
-	InstanceInterfaceNames []string `yaml:"instance_interface_names"`
-	InstancePort           int      `yaml:"instance_port"`
-	InstanceAddr           string   `yaml:"instance_addr"`
+	InstanceID             string   `yaml:"instance_id" doc:"hidden"`
```

Comment:
Why are these hidden?

Reply:
I marked them as hidden because I've the feeling they introduce confusion to the user. If you see them you may think you're expected to configure them, while they're here to cover very edge cases (ie. local dev env) and the default should work in most of the setups. I would re-introduce them, but I've the feeling they will just make the config file reference harder to understand when it comes to configuring the ring (same reason why they've been hidden in `ring.LifecyclerConfig`).